agent/agent-bootstrap/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpServerMetrics.java

```diff
@@ -0,0 +1,178 @@
+
+  private static final Logger logger = Logger.getLogger(HttpServerMetrics.class.getName());
+
+  /**
+   * Returns a {@link OperationMetrics} which can be used to enable recording of {@link
+   * HttpServerMetrics} on an {@link
+   * io.opentelemetry.instrumentation.api.instrumenter.InstrumenterBuilder}.
+   */
+  public static OperationMetrics get() {
+    return HttpServerMetrics::new;
+  }
+
+  private final LongUpDownCounter activeRequests;
+  private final DoubleHistogram duration;
+  private final LongHistogram requestSize;
+  private final LongHistogram responseSize;
+
+  private HttpServerMetrics(Meter meter) {
+    activeRequests =
+        meter
+            .upDownCounterBuilder("http.server.active_requests")
+            .setUnit("requests")
+            .setDescription("The number of concurrent HTTP requests that are currently in-flight")
+            .build();
+
+    duration =
+        meter
+            .histogramBuilder("http.server.duration")
+            .setUnit("ms")
+            .setDescription("The duration of the inbound HTTP request")
+            .build();
+    requestSize =
+        meter
+            .histogramBuilder("http.server.request.size")
+            .setUnit("By")
+            .setDescription("The size of HTTP request messages")
+            .ofLongs()
+            .build();
+    responseSize =
+        meter
+            .histogramBuilder("http.server.response.size")
+            .setUnit("By")
+            .setDescription("The size of HTTP response messages")
+            .ofLongs()
+            .build();
+  }
+
+  @Override
+  public Context onStart(Context context, Attributes startAttributes, long startNanos) {
+    activeRequests.add(1, TemporaryMetricsView.applyActiveRequestsView(startAttributes), context);
+
+    return context.with(
+        HTTP_SERVER_REQUEST_METRICS_STATE,
+        new AutoValue_HttpServerMetrics_State(startAttributes, startNanos));
+  }
+
+  @Override
+  public void onEnd(Context context, Attributes endAttributes, long endNanos) {
+    State state = context.get(HTTP_SERVER_REQUEST_METRICS_STATE);
+    if (state == null) {
+      logger.log(
+          FINE,
+          "No state present when ending context {0}. Cannot record HTTP request metrics.",
+          context);
+      return;
+    }
+    activeRequests.add(
+        -1, TemporaryMetricsView.applyActiveRequestsView(state.startAttributes()), context);
+    Attributes durationAndSizeAttributes =
+        TemporaryMetricsView.applyServerDurationAndSizeView(state.startAttributes(), endAttributes);
+
+    // START APPLICATION INSIGHTS CODE
+
+    // this is needed for detecting telemetry signals that will trigger pre-aggregated metrics via
+    // auto instrumentations
+    Span.fromContext(context).setAttribute(IS_PRE_AGGREGATED, true);
+
+    Attributes durationAttributes =
+        durationAndSizeAttributes.toBuilder()
+            .put(IS_SYNTHETIC, isUserAgentBot(endAttributes, state.startAttributes()))
```

Comment:
I think this would look nice, and avoids static importing the method (which can be tricky to read in github b/c there's no context, which is why tend to limit static imports to "known" commonly imported method/constants
```suggestion
            .put(IS_SYNTHETIC, UserAgents.isBot(endAttributes, state.startAttributes()))
```